### PR TITLE
Mjmlconfig

### PR DESCRIPTION
**Pull request by kmcb777**
Need doc update. Works the same as mjml3

### CODE REVIEW

File: packages/mjml/src/index.js [hunk 1]
```diff
@@ -46,4 +49,19 @@ registerComponent(CarouselImage)
...
+try {
+  const mjmlConfig = fs.readFileSync(path.join(process.cwd(), '.mjmlconfig'));
+  const custom_comps = JSON.parse(mjmlConfig).packages
+
+  custom_comps.forEach((compPath) => {
+    const requiredComp = require(path.join(process.cwd(), compPath))
+    registerComponent(requiredComp.default || requiredComp)
+  })
+} catch(e) {
+  if (e.code !== 'ENOENT') {
```
Comment (iRyusa) [inline packages/mjml/src/index.js]: We might need a better error handling for each possible cases ?

File: packages/mjml/src/index.js [hunk 2]
```diff
@@ -46,4 +49,19 @@ registerComponent(CarouselImage)
 
 registerDependencies(require('./dependencies'))
 
+
+try {
+  const mjmlConfig = fs.readFileSync(path.join(process.cwd(), '.mjmlconfig'));
+  const custom_comps = JSON.parse(mjmlConfig).packages
```
Comment (iRyusa) [inline packages/mjml/src/index.js]: camelCase 👎

**Comment by iRyusa** [comment]
Anyway, this should be done on `mjml-core` not on the community `mjml` package

File: packages/mjml/src/index.js (hunk 2 above)
Comment (kmcb777) [reply packages/mjml/src/index.js]: oups

File: packages/mjml/src/index.js (hunk 1 above)
Comment (kmcb777) [reply packages/mjml/src/index.js]: Actually ENOENT is the case where .mjmlconfig file does not exist
In this case we don't want to consider this as an error
For any other error we log it, but i haven't identified specific error cases

**Comment by kmcb777** [comment]
moved to mjml-core/index
please forgive my ignorance

**Comment by jskrzypek** [comment]
Shouldn't this functionality be run above, inside the `mjml2html()` function? 

The vscode-mjml extension is expecting it to respect passing an alternate directory as the filePath option for the mjml function.

I can make a PR if you want.

**Comment by iRyusa** [comment]
File path is only for relative include. You might need an option for mjmlconfig path ?

**Comment by jskrzypek** [comment]
Yeah I've been ill the past few days so i haven't had a chance to advance on this, but I realized over the w/e that I forgot to allow for node-modules. I'll update my branch/fork when I'm better.

**Comment by kmcb777** [comment]
@jskrzypek i'll merge this one, you can open a PR into the `next` branch when you'll update your branch 👍

**Pull request merged**